rest.go

```diff
@@ -90,39 +51,39 @@ func BuildResponse(res *http.Response) (*Response, error) {
 }
 
 // API is the main interface to the API.
-func API(request Request) (*Response, error) {
+func API(request *http.Request) (*Response, error) {
 	return DefaultClient.API(request)
 }
 
 // The following functions enable the ability to define a
 // custom HTTP Client
 
 // MakeRequest makes the API call.
-func (c *Client) MakeRequest(req *http.Request) (*http.Response, error) {
+func (c *Client) makeRequest(req *http.Request) (*http.Response, error) {
 	return c.HTTPClient.Do(req)
 }
 
 // API is the main interface to the API.
-func (c *Client) API(request Request) (*Response, error) {
-	// Add any query parameters to the URL.
-	if len(request.QueryParams) != 0 {
-		request.BaseURL = AddQueryParameters(request.BaseURL, request.QueryParams)
+func (c *Client) API(request *http.Request) (*Response, error) {
+	if request == nil {
+		return nil, errors.New("invalid request: request cannot be nil")
 	}
+	_, supportedMethod := supportedMethods[request.Method]
 
-	// Build the HTTP request object.
-	req, err := BuildRequestObject(request)
-	if err != nil {
-		return nil, err
+	if !supportedMethod {
+		return nil, errors.New("unsupported request method:" + request.Method)
+	}
+
+	if request.Header.Get("Content-Type") == "" && request.Body != nil {
+		request.Header.Set("Content-Type", "application/json")
 	}
```

Comment:
request is actually of type http.Request so there's no control over the type of request.Body as it's a builtin. I only added this handling because there was an existing use case where content type is supposed to default to application/json if it's not set and there is a request body.